modules/flowable-cmmn-rest/src/main/java/org/flowable/cmmn/rest/service/api/CmmnRestResponseFactory.java

```diff
@@ -597,7 +597,18 @@ public VariableInstanceResponse createVariableInstanceResponse(VariableInstance
             result.setCaseInstanceUrl(urlBuilder.buildUrl(CmmnRestUrls.URL_CASE_INSTANCE, variableInstance.getScopeId()));
         }
         result.setTaskId(variableInstance.getTaskId());
-        result.setVariable(createRestVariable(variableInstance.getName(), variableInstance.getValue(), null, variableInstance.getId(), VARIABLE_VARINSTANCE, false, urlBuilder));
+
+        RestVariableScope scope = RestVariableScope.GLOBAL;
+        boolean localCmmn = variableInstance.getSubScopeId() != null && !variableInstance.getSubScopeId().equals(variableInstance.getScopeId()); // cmmn
+        boolean taskLocal = variableInstance.getTaskId() != null;
+
+        if (localCmmn || taskLocal) {
+            scope = RestVariableScope.LOCAL;
+        }
```

Comment:
Instead of doing lit like this we can implement this in the following way:

```suggestion
        RestVariableScope scope;
        if (variableInstance.getSubScopeId() != null && !variableInstance.getSubScopeId().equals(variableInstance.getScopeId()) ||  variableInstance.getTaskId() != null) {
            scope = RestVariableScope.LOCAL;
        } else {
            scope = RestVariableScope.GLOBAL;
        }
```

If it is easier for you you can also extract a method that will do this logic in the if statements